copy.go

```diff
@@ -86,66 +87,38 @@ func copyNextOrSkip(src, dest string, info os.FileInfo, opt Options) error {
 // with considering existence of parent directory
 // and file permission.
 func fcopy(src, dest string, info os.FileInfo, opt Options) (err error) {
-
-	var readcloser io.ReadCloser
-	if opt.FS != nil {
-		readcloser, err = opt.FS.Open(src)
-	} else {
-		readcloser, err = os.Open(src)
-	}
-	if err != nil {
-		if os.IsNotExist(err) {
-			return nil
-		}
-		return
-	}
-	defer fclose(readcloser, &err)
-
 	if err = os.MkdirAll(filepath.Dir(dest), os.ModePerm); err != nil {
 		return
 	}
 
-	f, err := os.Create(dest)
-	if err != nil {
-		return
+	// Use FileCopyMethod to do copy.
+	if err = opt.FileCopyFunc.fcopy(src, dest, info, opt); err != nil {
+		if pathError, ok := err.(*os.PathError); ok && errors.Is(err, os.ErrNotExist) && pathError.Path == src {
```

Comment:
Just at a glance, it looks like increasing complexity of source code.
Do we really have a specific issue we need to solve in this PR?
or do you think we can separate PR for different point of interest?

Reply:
If I don't do this here, it breaks the `file is deleted while copying` test. Let me see if I can do something else though

Reply:
I changed it to have a `skipFile` return instead